### DISCUSSION TITLE
Is there a way for freshrss to force update a feed if it errors out once?

### DISCUSSION THREAD

**Discussion by neilstottler** [FreshRSS/FreshRSS#6980] [Q&A]
Hi I have all my feeds but some error out and then never update again. When I run the update manually they update fine. Curious how I can tell freshrss to force update even if it fails once.

**Comment by johnpc** [comment]
Did you ever figure out how to do this?

**Reply by johnpc** [reply]
For what it's worth I couldn't find any button in the UI allowing this apart from manually clicking the gear next to every failed subscription and clicking retry (which is not practical for me since it's a lot of feeds!). I even tried running `./app/www/app/actualize_script.php` via docker exec, but that also skipped the failed feeds.

Instead I found that you can open chrome's dev tools and run this script:

```
const MAX_FEED_ID=1000;
async function refreshAllFeeds() {
    for (let id = 0; id <= MAX_FEED_ID; id++) {
        try {
            const response = await fetch(`<YOUR_FRESHRSS_DOMAIN>/i/?c=feed&a=actualize&id=${id}`);
            console.log(`Fetched ID: ${id}, Status: ${response.status}`);
            
            // Optional: Add a small delay to avoid overwhelming the server
            await new Promise(resolve => setTimeout(resolve, 500));
            
        } catch (error) {
            console.error(`Error fetching ID ${id}:`, error);
        }
    }
}

refreshAllFeeds();
```

**Comment by Alkarex** [comment]
> how I can tell freshrss to force update even if it fails once

FreshRSS updates all feeds independently of whether there are in error or not. Check your FreshRSS logs for more information about those specific feeds.

There might be some behaviours such as your IP being blocked if you update some feeds too often, or too many feeds from the same, or something else.

**Comment by timfoster** [comment]
I'm not sure how to do this from the GUI, but my instance is backed by an sqlite3 database, and this _seems_ to clear the error state:
```
$ echo 'update feed set error=0;' | sqlite3 FreshRSS/data/users/timf/db.sqlite
```

Then reload the browser page, and hit `q` to refresh all feeds.

**Reply by Alkarex** [reply]
I would be curious to have more details as to why this should be needed at all.
The error state is not persisted. The feed is tried again at the next refresh opportunity independently of its error state.

**Reply by timfoster** [reply]
The error does seem to persist here? I can reproduce by intentionally breaking a feed URL (e.g. appending "xxx" to it) and refreshing with `q`

After refreshing, we get the warning triangle beside the feed section, and the `Blast! This feed has encountered a problem. If this situation persists, please verify that it is still reachable.` error message on the feed management page.

If we then fix the url, and use `q` to refresh, the error state remains. Checking the sqlite db, is still has the `error` boolean set:

```
sqlite> select error from feed where name = 'BMW BLOG';
1
sqlite> 
```

Is there some other way to force a refresh other than `q` (which I think was the original question in this thread)

**Reply by Alkarex** [reply]
The feed (in error or not) will not actually be refreshed until the next time it is allowed to do so based on its refresh policy.
So let's say you have a policy of refreshing once per hour, you need to wait till the first refresh after that, for that feed to be included in a global refresh.
Note that some feeds may have additional caching rules (sent by HTTP).

**Reply by Frenzie** [reply]
Otoh I think the "Update feeds" button in the individual feed settings does it immediately, doesn't it?

**Reply by timfoster** [reply]
Ok, this makes sense now that you explain it. I wonder could we make it clearer to users somehow? Something like:

"Blast! This feed has encountered a problem. FreshRSS will automatically attempt to refresh again according to the feed schedule. If this situation persists, please verify that it is still reachable."
